### ISSUE TITLE
[Feat]: Upgrade vscode to 1.70

### ISSUE THREAD

**Issue by bala** [coder/code-server#5417]
## What is your suggestion?
Upgrade vscode to 1.70

**Comment by fritterhoff** [comment]
![grafik](https://user-images.githubusercontent.com/32478819/183578860-05e8d245-5bef-45a2-951c-a001deb2e24a.png)

So some first steps (#5422) towards the 1.70 look fine. Jupyter is working, `code-server` in the cli looks also good to me.

**Comment by fritterhoff** [comment]
@benz0li feel free to have a look on the latest CI artifacts if you can detect some odd bugs. (https://github.com/coder/code-server/actions/runs/2815650333) 😉

**Comment by benz0li** [comment]
> @benz0li feel free to have a look on the latest CI artifacts if you can detect some odd bugs.

Will do.

**Comment by benz0li** [comment]
I have deployed the latest latest CI artifact for `linux-amd64` at https://vscode-r.jupyter.b-data.ch, Image `R (base:test) + code-server`.

**Comment by benz0li** [comment]
> Jupyter is working

Confirmed.

*  Python v2022.12.0
*  Jupyter v2022.8.1002170547

**Comment by benz0li** [comment]
> `code-server` in the cli looks also good to me.

JupyterLab Terminal:

```zsh
╰─ code-server --version
4.5.1 c5c16630adb086bf2cfed1ba6d678ef2ac9d06e1 with Code 1.70.1
```

```zsh
╰─ echo $PATH
/home/benz0li/.local/bin:/opt/code-server/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
```

```zsh
╰─ which code-server
/opt/code-server/bin/code-server
```

code-server Terminal:

```zsh
╰─ code-server --version
1.70.1
2062a59ca1a586d8a6e7bf483841085a94c440a4
x64
```

```zsh
╰─ echo $PATH
/opt/code-server/lib/vscode/bin/remote-cli:/home/benz0li/.local/bin:/opt/code-server/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
```

```zsh
╰─ which code-server
/opt/code-server/lib/vscode/bin/remote-cli/code-server
```

I can live with this :wink:. Related to https://github.com/coder/code-server/issues/5335#issuecomment-1189928660 ff.

**Comment by bala** [comment]
@benz0li @jsjoeio @code-asher can we publish this build as beta and test the naturally resolved bugs like #5335 and close it?

**Comment by benz0li** [comment]
ℹ️ I am using

```zsh
# set PATH so it includes user's private bin if it exists
if [ -d "$HOME/bin" -a "$SHLVL" = 1 -a ! "$TERM_PROGRAM" = "vscode" ] ; then
    PATH="$HOME/bin:$PATH"
fi

# set PATH so it includes user's private bin if it exists
if [ -d "$HOME/.local/bin" -a "$SHLVL" = 1 -a ! "$TERM_PROGRAM" = "vscode" ] ; then
    PATH="$HOME/.local/bin:$PATH"
fi
```

in `~/.zshrc` to get identical `PATH` handling in JupyterLab and code-server, both for login and no-login shells as well as nested `zsh`, `screen` and `tmux` sessions.

**Comment by fritterhoff** [comment]
@benz0li great. Thanks for testing. Two features/patches required some more adjustments. One is regarding the language packs, the other is about settings storage. From my point of view both patches/features should work correctly at the moment but I'm not sure.

**Comment by benz0li** [comment]
> @benz0li @jsjoeio @code-asher can we publish this build as beta

I can't say anything about that because I do not work @coder.

But if a two-month rhythm is followed, [at least] one release candidate would be a good idea – for in-depth testing before the actual release.

> and test the obviously resolved bugs like #5335 and close it?

You may test at https://vscode-r.jupyter.b-data.ch with Image `R (base:test) + code-server`.  
ℹ️ I have whitelisted your GitHub account.

**Comment by jsjoeio** [comment]
side note: we had an issue with publishing a beta version to npm. we just fixed that. 

```shell
yarn global add code-server@beta
```

will get you the latest off `main`

https://www.npmjs.com/package/code-server/v/4.5.1-beta-3335d0a456da0ea639b1e010aeb9801a876a9323

**Comment by benz0li** [comment]
Functionality modified by patches tested and found to work:

* base-path
* cli-window-open
* local-storage
    * I assume this is about settings
* marketplace
* proxy-url
* service-worker
* webview

**Comment by benz0li** [comment]
Menu entry `Sign out of code-server` is available although `code-server` was started with `--auth none`.

**Comment by benz0li** [comment]
@fritterhoff Regarding https://github.com/coder/code-server/issues/5417#issuecomment-1208964440: What does `echo $PATH` and `which code-server` return in `code-server` Terminal?

**Comment by fritterhoff** [comment]
> @fritterhoff Regarding [#5417 (comment)](https://github.com/coder/code-server/issues/5417#issuecomment-1208964440): What does `echo $PATH` and `which code-server` return in `code-server` Terminal?

Since https://github.com/coder/code-server/pull/5360 fixes the script it self and not the way it is called the `$PATH` modification and output are produced by the official script and not by the modified `code-server` script.

**Comment by benz0li** [comment]
> Since #5360 fixes the script it self and not the way it is called the `$PATH` modification and output are produced by the official script and not by the modified `code-server` script.

Yours is not the output of the official script. That is why I am asking.

**Comment by fritterhoff** [comment]
Whooopss... that was to simple 😁 
A `which code-server` produces a `/usr/bin/code-server`, a `echo $PATH`
```
echo $PATH
/usr/lib/code-server/lib/vscode/bin/remote-cli:/opt/conda/bin:/opt/conda/condabin:/opt/conda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
```

**Comment by fritterhoff** [comment]
In addition: 
```
(base) jovyan@jupyter-ritterho-40hm-2eedu:/usr/lib/code-server/lib/vscode/bin/remote-cli$ ls /usr/lib/code-server/lib/vscode/bin/remote-cli
code.cmd  code-darwin.sh  code-linux.sh
(base) jovyan@jupyter-ritterho-40hm-2eedu:/usr/lib/code-server/lib/vscode/bin/remote-cli$ 
```

**Comment by benz0li** [comment]
> There is no code-server script only a code ... I have no idea whats wrong here 🤔

Maybe https://github.com/coder/code-server/issues/5335#issuecomment-1191025565? Do you still delete the `code-server` link/script?

In my case there is a link from `code-server` to `code-linux.sh`

**Comment by fritterhoff** [comment]
Ah your right. My Dockerfile contains a `rm /usr/lib/code-server/lib/vscode/bin/remote-cli/code-server`

**Comment by code-asher** [comment]
I wanted to wait until getting the 1.70 update in but in order to get https://github.com/coder/code-server/pull/5442 released I think we should make a release candidate today and then a regular release pretty soon after.

**Comment by code-asher** [comment]
> Menu entry Sign out of code-server is available although code-server was started with --auth none
Nice find!  I opened an issue and will fix it real swift https://github.com/coder/code-server/issues/5448

**Comment by xiaoyun94** [comment]
3 WAY MERGE VIA CODE SERVER

**Comment by jsjoeio** [comment]
Completed via https://github.com/coder/code-server/pull/5422

**Comment by benz0li** [comment]
> > Menu entry Sign out of code-server is available although code-server was started with --auth none
> 
> Nice find! I opened an issue and will fix it real swift #5448

I can confirm this is now fixed.

P.S.: I have deployed v4.6.0-rc.1 at https://vscode-r.jupyter.b-data.ch, Image `R (base:test) + code-server`.
